aws-cpp-sdk-core/source/external/cjson/cJSON.cpp

```diff
@@ -1018,7 +1018,7 @@ static cJSON_AS4CPP_bool print_string_ptr(const unsigned char * const input, pri
                     break;
                 default:
                     /* escape and print as unicode codepoint */
-                    sprintf((char*)output_pointer, "u%04x", *input_pointer);
+                    snprintf((char*)output_pointer, sizeof(output_pointer), "u%04x", *input_pointer);
```

Comment:
this is not correct

`output_pointer` is not an array on the stack like `char output_pointer[8]` that you can simply do `sizeof()` on

I don't know what it is without a bunch of investigation...

Reply:
Is the fix as simple as `(*output_pointer)`?

Reply:
Fixed with `output_buffer->length - (output_pointer - output_buffer->buffer)`
https://github.com/DaveGamble/cJSON/commit/ae49da2b61a378b9c0e07572ca759cd305c8b9c8